### PR TITLE
Fix #8590: Wallet Swap Service Refactor also Bump Core to 1.63.90

### LINKED ISSUE #8590: Wallet: Swap Service refactor in 1.63
iOS needs some refactor in swap service after core change merged https://github.com/brave/brave-core/pull/21406

### PR DESCRIPTION
**Pull request by nuo-xu**
## Summary of Changes
Core bump to 1.63.90
Swap Store class refactor 

This pull request fixes #8590

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Comment by soner-yuksel** [comment]
@nuo package-lock.json difference shouldn't be necessary. Did you use npm `install` or npm `update` ?

File: Tests/BraveWalletTests/SwapTokenStoreTests.swift [hunk 1]
```diff
@@ -247,8 +247,24 @@ class SwapStoreTests: XCTestCase {
...
     let swapService = BraveWallet.TestSwapService()
-    swapService._priceQuote = { $1(.init(), nil, "") }
-    swapService._transactionPayload = { $1(.init(), nil, "") }
+    swapService._quote = { _, completion in
+      if coin == .eth {
+        completion(.init(zeroExQuote: .init()), nil, "")
+      } else if coin == .sol {
+        completion(.init(jupiterQuote: .init()), nil, "")
+      } else {
+        completion(nil, nil, "Test Internal Error")
```
Comment (StephenHeaps) [inline Tests/BraveWalletTests/SwapTokenStoreTests.swift]: We should use `XCTFail` here instead

File: Tests/BraveWalletTests/SwapTokenStoreTests.swift [hunk 2]
```diff
@@ -247,8 +247,24 @@ class SwapStoreTests: XCTestCase {
...
+      }
+    }
+    swapService._transaction = { _, completion in
+      if coin == .eth {
+        completion(.init(zeroExTransaction: .init()), nil, "")
+      } else if coin == .sol {
+        completion(.init(jupiterTransaction: .init()), nil, "")
+      } else {
+        completion(nil, nil, "Test Internal Error")
```
Comment (StephenHeaps) [inline Tests/BraveWalletTests/SwapTokenStoreTests.swift]: Similar here, use `XCTFail`

**Review by StephenHeaps — APPROVED** [review]
2 minor nits

**Review by StephenHeaps — CHANGES_REQUESTED** [review]
I forgot the `package-lock.json` changes, we should resolve that before merge 🙂

**Review by StephenHeaps — APPROVED** [review]

**Pull request merged**